backend/app/main.py

```
@@ -479,153 +461,84 @@ async def get_mcp_logs():
-            },
-            "success": False,
-            "error": str(e)
-        }
+        logger.error(f"Error initiating crawl for job ID {job_id}: {str(e)}", exc_info=True)
+        # Update job status using the manager
+        update_overall_status(job_id=job_id, status='error', error_message=f"Error initiating crawl: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Error initiating crawl: {str(e)}")
+
+
+@app.get("/api/crawl-status/{job_id}", response_model=CrawlJobStatus)
+async def get_crawl_status(job_id: str = FastApiPath(..., title="Job ID", description="The unique ID of the crawl job")):
```

Ruff (0.8.2)

Missing return type annotation for public function `get_crawl_status`

(ANN201)